src/main/java/com/elixr/poc/rest/controller/PurchaseCreationController.java

```diff
@@ -0,0 +1,21 @@
+package com.elixr.poc.rest.controller;
+import com.elixr.poc.rest.request.PurchaseRequest;
+import com.elixr.poc.service.PurchaseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+/*controller class for create purchase using post mapping*/
+
+@RestController
+@RequestMapping("/controller")
+public class PurchaseCreationController {
+    @Autowired
+    PurchaseService purchaseService;
+
+    @PostMapping("/purchase")
+    public ResponseEntity<?> createPurchase(@RequestBody PurchaseRequest newPurchase) {
+        PurchaseService newService = new PurchaseService();
+        return newService.editResponse(newPurchase);
```

Comment:
I would suggest renaming the editResponse method to createPurchase or something like that. I think the purpose of the editResponse method is to create the purchase right?

Reply:
Yes.